### DISCUSSION TITLE
[Question] Flet web app static publish, browser shows blank screen

### DISCUSSION THREAD

**Discussion by psksvp** [flet-dev/flet#4487] [Q&A] [answered]
### Question

I have a web app which works fine  in dynamic mode (flet run --web --port 8000 main.py). It uses flet_map package. The app does not use any other package which pyodide does not support.  However when I try to publish as static website, the browser shows just blank screen. I tested it on both Firefox and Safari.

'flet build web --include-packages flet_map' produces the following log

![web1](https://github.com/user-attachments/assets/ee3cd859-ae96-4f02-89d0-056d9670170b)

'flet publish main.py" produces the following log

![web2](https://github.com/user-attachments/assets/ce2a4b58-2910-450f-949f-4f6ad36583b1)

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
- does this also happen with a hello world app?

- please share your requirements file and your file structure?

- what happens when you use [`flet build web`](https://flet.dev/docs/publish/web/static-website#flet-build-web)

**Reply by psksvp** [reply]
Yes, it does happen with the hello world (the counter example), the first screen shot of the my first post shows the log from running after `flet build web` 

* The content of requirements.txt is below

`flet==0.22.*`

* Below is the file structure

(.venv) psksvp@amoeba website % tree
.
├── README.md
├── __pycache__
│   ├── aboutPage.cpython-313.pyc
│   ├── consultationAndFeesPage.cpython-313.pyc
│   ├── contactPage.cpython-313.pyc
│   ├── homePage.cpython-313.pyc
│   ├── privacyPolicyPage.cpython-313.pyc
│   ├── repeatedElements.cpython-313.pyc
│   ├── servicesPage.cpython-313.pyc
│   ├── text.cpython-313.pyc
│   └── textAndImages.cpython-313.pyc
├── aboutPage.py
├── assets
│   ├── Down High Colour.jpg
│   ├── Down Low Colour.jpg
│   ├── Down Low Mono.jpg
│   ├── Logo.png
│   ├── Up High Mono .jpg
│   ├── Up Low Colour.jpg
│   ├── Up Low Mono.jpg
│   ├── accreditedPractisingDietitian.png
│   ├── aging.jpg
│   ├── cake.jpg
│   ├── gastrointestinalHealth.jpg
│   ├── heartHealth.jpg
│   ├── insulin.jpg
│   ├── menopause.jpg
│   ├── platterOfFood.jpg
│   ├── profile.jpg
│   ├── sambazon-qfIVXu7nxGU-unsplash.jpg
│   ├── sleapApnea.jpg
│   ├── strawberries.jpg
│   ├── text
│   │   ├── aboutPageBottom.txt
│   │   ├── aboutPageTop.txt
│   │   ├── consultationAndFees
│   │   │   ├── consultationInfo.txt
│   │   │   ├── description.txt
│   │   │   └── rebatesInfo.txt
│   │   ├── contact
│   │   │   ├── contactDescription.txt
│   │   │   ├── contactInfo.txt
│   │   │   ├── contactLindfield.txt
│   │   │   └── contactMonavale.txt
│   │   ├── homePageLeft.txt
│   │   ├── homePageRight.txt
│   │   ├── privacyPolicy.txt
│   │   └── services
│   │       ├── fullVersions
│   │       │   ├── gastrointestinalHealth.txt
│   │       │   ├── heartHealth.txt
│   │       │   ├── insulinResistance.txt
│   │       │   ├── nutritionalChallengesOfAging.txt
│   │       │   ├── pcos.txt
│   │       │   ├── perimenopause.txt
│   │       │   └── sleepApnoea.txt
│   │       └── shortenVersions
│   │           ├── gastrointestinalHealth.txt
│   │           ├── heartHealth.txt
│   │           ├── insulinResistance.txt
│   │           ├── nutritionalChallengesOfAging.txt
│   │           ├── pcos.txt
│   │           ├── perimenopause.txt
│   │           └── sleepApnoea.txt
│   └── veggies.jpg
├── build
├── consultationAndFeesPage.py
├── contactPage.py
├── homePage.py
├── main.py
├── privacyPolicyPage.py
├── repeatedElements.py
├── requirements.txt
├── servicesPage.py
├── text.py
└── textAndImages.py

**Comment by psksvp** [comment, marked as answer]
I've upgraded flet to 0.25.1 from 0.22 and `flet publish main.py` works.  Thanks.
